socketioxide/src/socket.rs

```diff
@@ -350,20 +352,24 @@ impl<A: Adapter> Socket<A> {
         &self.ns.path
     }
 
-    pub(crate) fn send(&self, mut packet: Packet) -> Result<(), Error> {
+    pub(crate) fn send(&self, mut packet: Packet) -> Result<(), SendError> {
         let payload = match packet.inner {
             PacketData::BinaryEvent(_, ref mut bin, _) | PacketData::BinaryAck(ref mut bin, _) => {
                 std::mem::take(&mut bin.bin)
             }
             _ => vec![],
         };
+        let sender: Sender<EnginePacket> = self.tx.clone();
+        let packet = packet.try_into()?;
+        Retryer::new(
+            self.sid,
+            sender.clone(),
```

Comment:
I think it would be much better to only spawn a retryer when there is a failure, it would avoid a useless tx clone :
```rust
    match self.tx.try_send(packet) {
            Ok(()) => Ok(()),
            Err(TrySendError::Closed(_)) => todo!(),
            Err(TrySendError::Full(packet)) => Err(RetryerError::Remaining(Retryer::new(
                self.sid,
                self.tx.clone(),
                Some(packet),
                payload.into(),
            )).into()),
        }
```

Reply:
but also it force to have almost the same code in two places, it will be harder to maintain changes of sending logic